### ISSUE TITLE
Keyboard not responding at the end of the unit

### ISSUE THREAD

**Issue by rg3** [yquake2/yquake2#57]
slackware64-current, SDL1.

After finishing the first unit, the keyboard is not responding and it's not possible to advance the game, apparently.

It's OK if you start to hate me now about all these keyboard issues.

Here's a quicksave to reproduce the issue: http://rg3.name/pub/quick.tar.gz

**Comment by Yamagi** [comment]
Heh. Actually I'm quite happy that someone is testing this stuff. It was a very intrusive change and I haven't had any time to play more than base1. :)

This time it's a classic Quake 2 construct: KeyEvent() sets a global variable 'anykeydown' if any key is pressed. The movement code reads this variable and if it's > 1 a special button event BUTTON_ANY is generated. This buttton event is written onto the network / local loop. The server sends it as part of the 'ucom' struct to ClientThink() were it's used to determine if the intermission can be left. In fact the intermission is the only consumer of this crazy construct.

The fix is quite easy, just increase anykeydown of a key is pressed and decrement it if a corresponding key up event is received. I'll commit it tomorrow.

**Comment by Yamagi** [comment]
And now it's fixed. Thank you. :)

**Comment by rg3** [comment]
Thanks. It appears there are still some bugs and weird things happening, but I'm working on a straightforward way to reproduce them.

**Comment by rg3** [comment]
Yeah, I think I got it. You need to use the previous quicksave.

* Launch the game.
* Quickly press the quickload shortcut barely letting the ID logo video play.
* The game loads, then the unit ends, then the next video is skipped, then the next level loads without user intervention.

**Comment by rg3** [comment]
Testing it around 20 times more reveals it's not always reproducible. It may have to do with uninitialized variables or something, more than timing.

**Comment by Yamagi** [comment]
Okay, another try. I've played through the first 2 units and at least there're no obvious bugs left. Which, of cource, doesn't mean that Quake II hasn't got any surprises and strange corner cases left. :)

**Comment by rg3** [comment]
I think I've found a way to reproduce the weird behavior I described before and which is still not fixed, but I'm not sure if it's related to my keyboard layout or if you'd be able to reproduce it (layout is "es"). In any case, this is an obscure situation and the keyboard seems to work fine without major issues. Anyway, here it goes:

To bring down the console while in game I have to press the [grave accent](https://en.wikipedia.org/wiki/Grave_accent) key twice. I don't know if that's a problem by itself or if it's supposed to work with a single key press.

So I'm playing the game, any level, and I bring the console down as I said (the issue doesn't happen by going to the console through the menu). Now, no matter how I go back to the game (double press on the grave accent key again, or Escape/Escape), and it doesn't matter either if after being back in the game I move a bit, or shoot or whatever, without apparent issues, if later I press the quick load key to load the quicksave from this issue, the game is loaded at the end of the unit intermission, the intermission finishes, the video is skipped and I'm in the next level without pressing anything after "quick load".
